python/tox.ini

```diff
@@ -0,0 +1,56 @@
+[tox]
+envlist = py{27,36}, coverage
+skip_missing_interpreters = true
+
+
+[testenv]
+passenv = APPVEYOR* CI SSL_CERT_FILE TEST_MARKER TRAVIS*
+setenv = COVERAGE_FILE={toxworkdir}/.coverage.{envname}
+usedevelop = True
+deps = -rrequirements.txt
+       -rtest/requirements.txt
+# TODO: way too many warnings in tests, remove `-p no:warnings` and get rid of them all
+commands = pytest {posargs:-vv -p no:warnings -m {env:TEST_MARKER:main} --doctest-modules --cov-report term-missing --cov=src/}
+#          python -m unittest discover -s test/ -p '*_test.py'
+
+[testenv:coverage]
+passenv = {[testenv]passenv}
+setenv = COVERAGE_FILE={toxworkdir}/.coverage
+basepython = python
+skip_install = True
+deps = coverage
+commands = coverage combine
+           coverage report -i -m
+           coverage xml -i -o "{toxworkdir}/coverage.xml"
+           coverage html -i -d "{toxworkdir}/coverage"
+
+
+# The 'venv*' sections are a handy way of creating a venv for local development (with python3 or python2)
+[testenv:venv]
+basepython = python3
+envdir = .venv
+usedevelop = True
+deps = -rrequirements.txt
+       -rtest/requirements.txt
+commands = python --version
+
+[testenv:venv2]
+basepython = python2
+envdir = .venv2
+usedevelop = True
+deps = -rrequirements.txt
+       -rtest/requirements.txt
+commands = python --version
+
+
+# Configuration
+[pytest]
+cache_dir = .tox/.cache
+python_files = *test.py
+markers =
+    main: Main tests (exludes 'extra' and 'lib')
+    extra: Exercise test/extra/ (requires ffmpeg installed)
+    lib: Exercise test/lib/ (requires testlib installed).
```

Comment:
These just define all the markers used, `pytest --markers` would show those 3 (with their description), it's like a relatively common practice to declare them like this.

pytest looks at the following file for configuration (in that order): `pytest.ini`, `tox.ini`, `setup.cfg`